src/visor/unconfirmed.go

```diff
@@ -414,50 +415,71 @@ func (utp *UnconfirmedTxnPool) GetKnown(txns []cipher.SHA256) coin.Transactions
 // SpendsForAddresses returns all unconfirmed coin.UxOut spends for addresses
 // Looks at all inputs for unconfirmed txns, gets their source UxOut from the
 // blockchain's unspent pool, and returns as coin.AddressUxOuts
-func (utp *UnconfirmedTxnPool) SpendsForAddresses(bcUnspent *coin.UnspentPool,
-	a map[cipher.Address]byte) coin.AddressUxOuts {
-	auxs := make(coin.AddressUxOuts, len(a))
+func (utp *UnconfirmedTxnPool) SpendsForAddresses(unspent *blockdb.UnspentPool,
+	addrs []cipher.Address) (coin.AddressUxOuts, error) {
+	addrm := make(map[cipher.Address]byte, len(addrs))
+	for _, addr := range addrs {
+		addrm[addr] = byte(1)
+	}
+
+	auxs := make(coin.AddressUxOuts, len(addrs))
 	if err := utp.Txns.forEach(func(_ cipher.SHA256, tx *UnconfirmedTxn) error {
 		for _, h := range tx.Txn.In {
-			if ux, ok := bcUnspent.Get(h); ok {
-				if _, ok := a[ux.Body.Address]; ok {
-					auxs[ux.Body.Address] = append(auxs[ux.Body.Address], ux)
-				}
+			ux, ok, err := unspent.Get(h)
+			if err != nil {
+				return err
+			}
+
+			if !ok {
+				// unconfirm transaction's IN is not in the unspent pool, this should not happen
```

Comment:
hmm, I think need to return a error to describe this situation, in case of unexpected error.